### ISSUE TITLE
Add a way to temporarily disable uBlock on a domain or tab

### ISSUE THREAD

**Issue by Destroy666x** [uBlockOrigin/uBlock-issues#2694]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

A feature to temporarily disable blocking of ads on a tab or domain is definitely missing and not solved yet. There were bunch of similar issues in this repo, but rather focused on disabling it in all tabs, also one for single tabs in old repo: https://github.com/gorhill/uBlock/issues/3178 that had no conclusion.

This is useful in many cases, for example:
- you open a website with anti-adblocker and it's an article that you'd like to read at that moment without thinking how to prevent the anti-adblocker
- you want to do research about ads that are on some websites, for various reasons

Adding and removing a website to trusted sites is more or less fine, but this could be more automated. You can easily take a break and forget to remove the website from the whitelist at a later point, then be spammed with ads once you open few tabs from the same website at a later point. I understand this is just a matter of max few clicks daily, but that's what QoL changes are for, to save thousands of users from doing these clicks.

### A specific URL where the issue occurs.

https://github.com/uBlockOrigin

### Steps to Reproduce

1. Go to a website.
2. Click on the extension menu.
3. There is no way to whitelist a website/tab temporarily.

### Expected behavior

There should be a way to whitelist a website/tab temporarily.

### Actual behavior

There is no way to whitelist a website/tab temporarily.

### uBO version

1.50

### Browser name and version

Chrome 114.0.5735.111

### Operating System and version

Windows 10 22H2

**Comment by gorhill** [comment]
> you open a website with anti-adblocker

Best is to report these to <https://github.com/uBlockOrigin/uAssets/issues> (use _chat_ icon in popup panel to make this easy), we deal with anti-content blocker with default settings/lists.

**Comment by krystian3w** [comment]
You have power button in uBo pop-up to disable filtering on page, works for 99% senario where you don't back to page quickly.

**Comment by gwarser** [comment]
Did there was ever a clear feature request to disable **per tab**?

**Comment by garry-ut99** [comment]
Another one: https://www.reddit.com/r/uBlockOrigin/comments/8i9q3j/temporarily_disable_on_a_site_without_adding_to/

**Comment by gorhill** [comment]
https://github.com/gorhill/uBlock/issues/3178

**Comment by Destroy666x** [comment]
> > you open a website with anti-adblocker
> 
> Best is to report these to https://github.com/uBlockOrigin/uAssets/issues (use _chat_ icon in popup panel to make this easy), we deal with anti-content blocker with default settings/lists.

Yeah, that's a solution for long-term. But as I mentioned in my examples, it doesn't quite help in the short-term.

**Comment by thehale** [comment]
In addition to ad-blocking, I also leverage uBlock's filtering rules to completely block websites I find myself spending too much time on (e.g. social media). A feature for temporary unblocking would be nice for one-offs (checking service status, being active on Twitter for a dev conference, etc.)

### UI Proposal
Perhaps the "on/off" button could have two options, one for a permanent toggle (same as current behavior), and one for a temporary toggle (the new proposed behavior). Here's a quick sketch up of how the current design (top row) could compare to a design that has the two-toggle functionality (bottom row):

![ublock origin temporary unblock](https://github.com/uBlockOrigin/uBlock-issues/assets/47901316/9c0308ea-34ab-4f76-8f01-de02fe044860)

### "Backend" Proposal
This could be implemented as an extension to the syntax for [Trusted Websites](https://github.com/gorhill/uBlock/wiki/How-to-mark-a-web-site-as-trusted), where an `until` key specifies the Unix Timestamp (in seconds) at which the trust expires.

```python
# current syntax - permanent trust
example.com

# proposed syntax extension - temporary trust
[until=1703495957] example.com
```

This syntax could also work for "permanent" trust by setting the `until` to a far-out timestamp (in this case the max-value for a signed 32-bit integer).
```python
# proposed syntax extension - alternate permanent trust
[until=2147483647] example.com
```

The usage of brackets for the key-value pairs should not interfere with the regex syntax since regex rules always start with a `/`
```
[until=1703495957] /^https://[0-9a-z-]+//
```

### Caveats
- This proposal does nothing to address blocking for a specific tab.
- Extra syntax means extra parsing logic
- It would probably also be good to add a Setting to customize the default "trust" time (instead of always 10 minutes)

**Comment by jimbobmcgee** [comment]
I would also very much like to have a temporary override for the current tab.  The lack of such a feature has indeed stopped me from migrating to uBO in the past (I am now biting the bullet and trialling uBO for...*other* reasons).

For all its sins, NoScript has exactly this feature -- you can click one button in the extension menu to disable NoScript in a specific tab, regardless of what it then goes on to load.  When you close the tab, all restrictions are re-enabled automatically.

It is incredibly useful for online stores that send you through payment processors, credit card verifications, real-human detection, anti-fraud, etc. that you would not normally know about until *after* they have failed.  Continual rinse-and-repeat/trial-and-error of the filters until the payment goes through is an exercise in tedium (especially if they drop your basket after sending you to the payment provider); as is having to explain that a double-charge should be refunded because you content-blocked part of their website so it couldn't work as "intended".

I'm not necessarily sold on @thehales's time-based approach, but the browser tab *is* a reasonable functional boundary to many end-users.  Sometimes you just need to get something done, *once*, and move on, and a one-tab override would certainly enable that.
